includes/class-newspack-popups-inserter.php

```diff
@@ -326,9 +329,63 @@ function ( $popup ) use ( $shortcoded_popups_ids ) {
 		foreach ( $overlay_popups as $overlay_popup ) {
 			$output = '<!-- wp:html -->' . Newspack_Popups_Model::generate_popup( $overlay_popup ) . '<!-- /wp:html -->' . $output;
 		}
+		return $output;
+	}
+
+	/**
+	 * Process popups and insert into post and page content if needed.
+	 *
+	 * @param string $content The content of the post.
+	 */
+	public static function insert_popups_in_content( $content = '' ) {
+		if (
+			// Avoid duplicate execution.
+			true === self::$the_content_has_rendered
+			// Not Frontend.
+			|| is_admin()
+			// Content is empty.
+			|| empty( trim( $content ) )
+			// No popup insertion in archive pages - there's another method for that.
+			|| ! is_singular()
+			// If not in the loop, ignore.
+			|| ! in_the_loop()
+			// Don't inject inline popups on paywalled posts.
+			// It doesn't make sense with a paywall message and also causes an infinite loop.
+			|| function_exists( 'wc_memberships_is_post_content_restricted' ) && wc_memberships_is_post_content_restricted()
+		) {
+			return $content;
+		}
+
+		// If any popups are inserted using a shortcode, skip them - no need to duplicate.
+		$shortcoded_popups_ids = self::get_shortcoded_popups_ids( get_the_content() );
+		$popups                = array_filter(
+			self::popups_for_post(),
+			function ( $popup ) use ( $shortcoded_popups_ids ) {
+				return ! in_array( $popup['id'], $shortcoded_popups_ids ) && Newspack_Popups_Model::should_be_inserted_in_page_content( $popup );
+			}
+		);
+
+		if ( empty( $popups ) ) {
+			return $content;
+		}
+
+		if ( function_exists( 'scaip_maybe_insert_shortcode' ) ) {
+			// Prevent default SCAIP insertion.
+			remove_filter( 'the_content', 'scaip_maybe_insert_shortcode', 10 );
```

Comment:
With the new approach for detecting `core/heading` and aligned `core/image`, maybe we can do the same with SCAIP inserted content and preserve how the SCAIP filter is applied?

It would be better to maintain since we would be parsing another plugin inserted content instead of manually applying its logic.

Reply:
> With the new approach for detecting core/heading and aligned core/image, maybe we can do the same with SCAIP inserted content 

You mean add the same changes as this PR introduces to SCAIP?

> It would be better to maintain since we would be parsing another plugin inserted content instead of manually applying its logic.

Actually with the block-insertion change to SCAIP this manual triggering of SCAIP is no longer necessary. This block was here to prevent SCAIP from inserting ads mid-prompt, but it won't do that with the block-based insertion. I've removed the block in 7952d33e282a8b12f11f3d1a4e4b9066fff0adaf

Reply:
> You mean add the same changes as this PR introduces to SCAIP?

No, I meant to use the `can_block_be_followed_by_prompt` to also check for SCAIP instead of modifying the filter. My guess is the `render_block()` returned here from a filter with priority 1 was also messing with the `parse_bock()` in SCAIP, which is priority 5. That would call for a manual triggering as well.